sympy/solvers/solveset.py

```diff
@@ -231,6 +235,26 @@ def _invert_complex(f, g_ys, symbol):
                                for g_y in g_ys if g_y != 0])
             return _invert_complex(f.args[0], exp_invs, symbol)
 
+    if isinstance(f, TrigonometricFunction):
```

Comment:
Let's move this to `_solve_trig` -- it's a special case of solving a trig expression: the case when the expression *is* a trig function. These cases can be solved without rewriting as exp.